### ISSUE TITLE
Switching input causes crash problem

### ISSUE THREAD

**Issue by hotic** [Korea-Minecraft-Forum/CocoaInput#4]
While using the latest build of CocoaInput 4.3.1 fabric on MacOS 14.0 Sonama Beta, switching the input method results in a game crash. 
My JVM version is Zulu JDK 17.0.8, and I did not encounter this issue on MacOS Ventura 13.5.1.

GIF and error report:

https://imgur.com/kYKgcYF

https://gist.github.com/hotic/a0b6c9377744adaba844eab7efa75b54

**Comment by LemonCaramel** [comment]
Hello, does macOS 14 RC have the same crash issue?
I upgraded to RC today, but there is no crash.

**Comment by ipiiia** [comment]
> Hello, does macOS 14 RC have the same crash issue? I upgraded to RC today, but there is no crash.

저도 저사람이랑 동일한 문제가 있어요 한글이랑 영어는 딱히 크래쉬가 안나는거같고
게임내에서 일본어같은 IME를 사용하는 언어로 변경하거나(컨트롤스페이스 대신에 언어변경키로 변경하면 더 높은확률로 충돌) 펑션키로 운영체제 볼륨을 조절하거나 하는 운영체제 관련된 어떤 작업을 하면 자바가 죽어버리네요
카라멜, 코코아 모드가 없으면 튕기지않습니다

코코아나 카라멜이나 똑같이 튕기는걸보면
코코아라이브러리랑 소노마 관련돼서 뭔가 문제가있나봐요

caramelchat14392102543130322870.tmp<카라멜챗

libcocoainput.dylib<코코아인풋

**Comment by hotic** [comment]
> Hello, does macOS 14 RC have the same crash issue? I upgraded to RC today, but there is no crash.

It hasn't been fixed yet. 
upon my testing, the same issue exists on Mac OS Sonoma 14.0 (23A339), and the problem is also observed on the latest version of caramelChat.

**Comment by LemonCaramel** [comment]
I have limited time to use the Mac. Therefore, I need additional information for quick fix.

Is Rect _(character conversion box)_ position displayed correctly above Minecraft in Ventura?
(In Sonoma, it seems to be displayed outside of Minecraft.)
![image](https://github.com/Korea-Minecraft-Forum/CocoaInput/assets/45729082/8e3dfa46-705d-4311-9f59-24aa0a657aa7)


and... I saw the API change of Sonoma. but, no changes were found that affected CocoaInput.
To be honest, I'd like to believe this is a bug in macOS.

**Comment by LemonCaramel** [comment]
I had an important exam and my time to use a mac was limited, so I got delayed a lot. I apologize for the delay.

I have updated the release to temporarily fix this issue.
https://github.com/Korea-Minecraft-Forum/CocoaInput/releases/tag/4.4.0-0
(I haven't tested it yet, but I think it will be significantly improved.)

**Comment by hotic** [comment]
> I had an important exam and my time to use a mac was limited, so I got delayed a lot. I apologize for the delay.
> 
> I have updated the release to temporarily fix this issue. https://github.com/Korea-Minecraft-Forum/CocoaInput/releases/tag/4.4.0-0 (I haven't tested it yet, but I think it will be significantly improved.)

Oh Thanks, I'll definitely try.
Over the past few months, I've rolled back my system to Mac OS Ventura 13.6.1, mainly because Sonoma just hasn't been cutting it for me.
For starters, I can no longer use CocoaInput, and I've run into some glitches with Bartender, which you can check out here: https://www.macbartender.com/Bartender5/ScreenCaptureMenuBarItem/. On top of that, there are a bunch of compatibility issues with other software. 
Basically, the instability of Sonoma has really thrown off my workflow.

**Comment by ipiiia** [comment]
> I had an important exam and my time to use a mac was limited, so I got delayed a lot. I apologize for the delay.
> 
> I have updated the release to temporarily fix this issue. https://github.com/Korea-Minecraft-Forum/CocoaInput/releases/tag/4.4.0-0 (I haven't tested it yet, but I think it will be significantly improved.)

오 sonoma로 테스트해봤는데 잘되는것같아요 혹시 1.20.1도 지원해주실수 있을까요?

**Comment by LemonCaramel** [comment]
I fixed the possibility of a crash as much as possible.
If the problem still persists, use [this guide](https://github.com/Korea-Minecraft-Forum/CocoaInput#troubleshooting-macos-140) to revert to the input behavior from before Sonoma.
and I updated the last release.
